test/TwigEnvironmentFactoryTest.php

```diff
@@ -191,83 +215,87 @@ public function invalidConfiguration()
      * @param mixed $config
-     * @param string $contains
+     * @throws LoaderError
      */
-    public function testRaisesExceptionForInvalidConfigService($config, $contains)
+    public function testRaisesExceptionForInvalidConfigService($config, string $contains): void
     {
-        $this->container->has('config')->willReturn(true);
-        $this->container->get('config')->willReturn($config);
+        $this->container->expects(self::once())->method('has')->with('config')->willReturn(true);
+        $this->container->expects(self::once())->method('get')->with('config')->willReturn($config);
         $factory = new TwigEnvironmentFactory();
 
         $this->expectException(InvalidConfigException::class);
         $this->expectExceptionMessage($contains);
-        $factory($this->container->reveal());
+        $factory($this->container);
     }
 
-    public function testUsesTimezoneConfiguration()
+    /**
+     * @throws LoaderError
+     */
+    public function testUsesTimezoneConfiguration(): void
     {
-        $tz = DateTimeZone::listIdentifiers()[0];
+        $tz     = DateTimeZone::listIdentifiers()[0];
         $config = [
             'twig' => [
                 'timezone' => $tz,
             ],
         ];
-        $this->container->has('config')->willReturn(true);
-        $this->container->get('config')->willReturn($config);
-        $this->container->has(TwigExtension::class)->willReturn(false);
-        $this->container->has(\Zend\Expressive\Twig\TwigExtension::class)->willReturn(false);
-        $this->container->has(ServerUrlHelper::class)->willReturn(false);
-        $this->container->has(\Zend\Expressive\Helper\ServerUrlHelper::class)->willReturn(false);
-        $this->container->has(UrlHelper::class)->willReturn(false);
-        $this->container->has(\Zend\Expressive\Helper\UrlHelper::class)->willReturn(false);
-        $factory = new TwigEnvironmentFactory();
-        $environment = $factory($this->container->reveal());
-        $fetchedTz = $environment->getExtension(CoreExtension::class)->getTimezone();
+        $this->container->expects(self::exactly(2))->method('has')->withConsecutive(
+            ['config'],
+            [TwigExtension::class],
+        )->willReturnOnConsecutiveCalls(true, false);
+        $this->container->expects(self::once())->method('get')->with('config')->willReturn($config);
+        $factory     = new TwigEnvironmentFactory();
+        $environment = $factory($this->container);
+        /** @var CoreExtension $extension */
+        $extension = $environment->getExtension(CoreExtension::class);
+        $fetchedTz = $extension->getTimezone();
         $this->assertEquals(new DateTimeZone($tz), $fetchedTz);
     }
 
-    public function testRaisesExceptionForInvalidTimezone()
+    /**
+     * @throws LoaderError
+     */
+    public function testRaisesExceptionForInvalidTimezone(): void
     {
-        $tz = 'Luna/Copernicus_Crater';
+        $tz     = 'Luna/Copernicus_Crater';
```

Comment:
Oh no. This made me realize we will have to deal with timezones on other planetary bodies in the next decade or two.

Reply:
woah